packet_handler_map.go

```diff
@@ -86,10 +87,10 @@ func (h *packetHandlerMap) Retire(id protocol.ConnectionID) {
 }
 
 func (h *packetHandlerMap) retireByConnectionIDAsString(id string) {
+	h.mutex.Lock()
```

Comment:
I don't think it's necessary to move the locking of the mutex here.

Reply:
The idea was that this would avoid a potential race condition; if `h.deleteRetiredSessionsAfter` is 0, the remove-function could run immediately and execute `delete(h.deletionTimers, id)` *before* the insertion happens, and therefore leak memory. Also, I can't think of any downsides to this; creating the timer should be fairly quick, so the lock is not held significantly longer.
If you're not convinced, I'm happy to move this back though.

Reply:
You’re right, thanks for the explanation. Since this is not obvious, could you maybe add a comment explaining why you lock the mutex there already?

Reply:
Done.